AGENTS.md

```diff
@@ -0,0 +1,271 @@
+    * distributed under the License is distributed on an "AS IS" BASIS,
+    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    * See the License for the specific language governing permissions and
+    * limitations under the License.
+    *
+    ***************************************************************/
+   ```
+
+2. **Package Naming**: Use lowercase package names matching the directory name
+
+3. **Error Handling**: Always check and handle errors appropriately
+
+4. **Testing**: Place test files alongside source files with `_test.go` suffix
+
+5. **Imports**: Use `goimports` for automatic import organization
+
+### TypeScript/React Code
+
+1. **Components**: Use functional components with hooks
+
+2. **Type Safety**: Prefer explicit types over `any`
+
+3. **File Organization**: 
+   - Components in `web_ui/frontend/components/`
+   - Pages/routes in `web_ui/frontend/app/`
+   - Utilities in `web_ui/frontend/helpers/`
+   - Custom hooks in `web_ui/frontend/hooks/`
+
+4. **Styling**: Use Material-UI components and styling system
+
+5. **Formatting**: Follow Prettier configuration (`.prettierrc.json`)
+
+## Development Workflow
+
+### Environment Setup
+
+Pelican uses Docker containers and VSCode Dev Containers for development. See [CONTRIBUTE.md](CONTRIBUTE.md) for detailed setup instructions.
+
+Key steps:
+1. Fork and clone the repository
+2. Install Docker and VSCode
+3. Pull the development container: `hub.opensciencegrid.org/pelican_platform/pelican-dev:latest-itb`
+4. Create `.devcontainer/devcontainer.json` using the template from `dev/devcontainer.json`
+5. Build Pelican from source inside the container
+
+### Pull Request Guidelines
+
+- Address a single concern with minimal changes
+- Add tests for new functionality or bug fixes
+- Update documentation in `docs/` folder if needed
+- Follow the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr)
+- At least one core contributor must review and approve changes
+- Run linters and tests before submitting
+
+### Configuration
+
+- Server configuration: YAML files (typically `pelican.yaml`)
+- Client configuration: Can use federation discovery URLs or explicit configuration
+- Parameters documented in `docs/parameters.yaml`
+
+## Key Features and Patterns
+
+### Object Transfer
+
+- Client commands: `pelican object get`, `pelican object put`, `pelican object copy`
+- Support for multiple protocols (HTTP, XRootD)
+- Checksum verification (MD5, SHA1, CRC32)
+- Resume capabilities for interrupted transfers
+
+### Federation Model
+
+- **Director**: Routes client requests to appropriate origins/caches
+- **Registry**: Manages namespace registrations
+- **Origins**: Serve data from storage backends
+- **Caches**: Provide distributed caching layer
+
+### Authentication
+
+- Token-based authentication with support for various issuers
+- OAuth integration via CILogon
+- Support for WLCG tokens
```

Comment:
Pelican supports both WLCG tokens and SciTokens. 

Here's the documentation for the WLCG token profile: https://github.com/WLCG-AuthZ-WG/common-jwt-profile/blob/master/profile.md

And here's the documentation for the SciTokens profile: https://scitokens.org/technical_docs/Claims

Please include these references.